### ISSUE TITLE
[DocDB] Segmentation fault in yb-master yb::tserver::LookupTabletPeer

### ISSUE THREAD

**Issue by shishir2001-yb** [yugabyte/yugabyte-db#21909]
Jira Link: [DB-10807](https://yugabyte.atlassian.net/browse/DB-10807)
### Description

Version: **2.23.0.0-b126**

Logs: [https://drive.google.com/file/d/1Z1R4No9eEIXwXmbp7E2QcChh05da9sMS/view?usp=sharing](https://drive.google.com/file/d/1Z1R4No9eEIXwXmbp7E2QcChh05da9sMS/view?usp=sharing)

Encountered Master coredump while running Cross DB DDL sample with DDL atomicity
Test Description:
```
1. Create a universe with DDL atomicity and Per DB Catalog mode
2. Run cross DB DDL sample app
```
Core dump(Find full coredump in the logs)
```
* thread #1, name = 'yb-master', stop reason = signal SIGSEGV
  * frame #0: 0x0000aaaad90f5954 yb-master`yb::tserver::LookupTabletPeer(yb::tserver::TabletPeerLookupIf*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) [inlined] yb::tablet::RaftGroupStatePB std::__1::__cxx_atomic_load[abi:v170002]<yb::tablet::RaftGroupStatePB>(__a=0x00000000000000b0, __order=acquire) at cxx_atomic_impl.h:363:12
    frame #1: 0x0000aaaad90f5954 yb-master`yb::tserver::LookupTabletPeer(yb::tserver::TabletPeerLookupIf*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) [inlined] std::__1::__atomic_base<yb::tablet::RaftGroupStatePB, false>::load[abi:v170002](this=0x00000000000000b0, __m=acquire) const at atomic_base.h:60:14
    frame #2: 0x0000aaaad90f5954 yb-master`yb::tserver::LookupTabletPeer(yb::tserver::TabletPeerLookupIf*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) [inlined] yb::tablet::TabletPeer::state(this=<unavailable>) const at tablet_peer.h:273:19
    frame #3: 0x0000aaaad90f5954 yb-master`yb::tserver::LookupTabletPeer(yb::tserver::TabletPeerLookupIf*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) [inlined] yb::Result<yb::tserver::TabletPeerTablet> yb::tserver::(anonymous namespace)::DoLookupTabletPeer<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>(tablet_manager=<unavailable>, tablet_id="00000000000000000000000000000000") at service_util.cc:281:56
    frame #4: 0x0000aaaad90f5910 yb-master`yb::tserver::LookupTabletPeer(tablet_manager=<unavailable>, tablet_id="00000000000000000000000000000000") at service_util.cc:306:10
    frame #5: 0x0000aaaad91291d4 yb-master`yb::tserver::TabletServiceImpl::UpdateTransaction(yb::tserver::UpdateTransactionRequestPB const*, yb::tserver::UpdateTransactionResponsePB*, yb::rpc::RpcContext) [inlined] yb::Result<yb::tserver::TabletPeerTablet> yb::tserver::LookupTabletPeerOrRespond<yb::tserver::UpdateTransactionResponsePB, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>(tablet_manager=<unavailable>, tablet_id=<unavailable>, resp=0x000027953f874758, context=0x0000ffff7b396900) at service_util.h:166:37
    frame #6: 0x0000aaaad91291cc yb-master`yb::tserver::TabletServiceImpl::UpdateTransaction(this=0x000027953f489020, req=0x000027953f874720, resp=0x000027953f874758, context=RpcContext @ 0x0000ffff7b396900) at tablet_service.cc:1189:24
    frame #7: 0x0000aaaad92d18e0 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) [inlined] yb::tserver::TabletServerServiceIf::InitMethods(this=<unavailable>, req=0x000027953f874720, resp=0x000027953f874758, rpc_context=RpcContext @ 0x0000ffff7b396850)::$_9::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>) const::'lambda'(yb::tserver::UpdateTransactionRequestPB const*, yb::tserver::UpdateTransactionResponsePB*, yb::rpc::RpcContext)::operator()(yb::tserver::UpdateTransactionRequestPB const*, yb::tserver::UpdateTransactionResponsePB*, yb::rpc::RpcContext) const at tserver_service.service.cc:883:9
    frame #8: 0x0000aaaad92d18b8 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) at local_call.h:126:7
    frame #9: 0x0000aaaad92d1508 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) [inlined] yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9::operator()(this=<unavailable>, call=<unavailable>) const at tserver_service.service.cc:881:7
    frame #10: 0x0000aaaad92d146c yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) [inlined] decltype(std::declval<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9&>()(std::declval<std::__1::shared_ptr<yb::rpc::InboundCall>>())) std::__1::__invoke[abi:v170002]<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9&, std::__1::shared_ptr<yb::rpc::InboundCall>>(__f=<unavailable>, __args=<unavailable>) at invoke.h:340:25
    frame #11: 0x0000aaaad92d1460 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) [inlined] void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:v170002]<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9&, std::__1::shared_ptr<yb::rpc::InboundCall>>(__args=<unavailable>, __args=<unavailable>) at invoke.h:415:5
    frame #12: 0x0000aaaad92d1460 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(std::__1::shared_ptr<yb::rpc::InboundCall>&&) [inlined] std::__1::__function::__alloc_func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()[abi:v170002](this=<unavailable>, __arg=<unavailable>) at function.h:192:16
    frame #13: 0x0000aaaad92d1460 yb-master`std::__1::__function::__func<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9, std::__1::allocator<yb::tserver::TabletServerServiceIf::InitMethods(scoped_refptr<yb::MetricEntity> const&)::$_9>, void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(this=<unavailable>, __arg=<unavailable>) at function.h:363:12
    frame #14: 0x0000aaaad92d8cc4 yb-master`yb::tserver::TabletServerServiceIf::Handle(std::__1::shared_ptr<yb::rpc::InboundCall>) [inlined] std::__1::__function::__value_func<void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()[abi:v170002](this=<unavailable>, __args=nullptr) const at function.h:517:16
    frame #15: 0x0000aaaad92d8ca8 yb-master`yb::tserver::TabletServerServiceIf::Handle(std::__1::shared_ptr<yb::rpc::InboundCall>) [inlined] std::__1::function<void (std::__1::shared_ptr<yb::rpc::InboundCall>)>::operator()(this=<unavailable>, __arg=nullptr) const at function.h:1168:12
    frame #16: 0x0000aaaad92d8ca8 yb-master`yb::tserver::TabletServerServiceIf::Handle(this=<unavailable>, call=<unavailable>) at tserver_service.service.cc:731:3
    frame #17: 0x0000aaaad8d7297c yb-master`yb::rpc::ServicePoolImpl::Handle(this=0x000027953f9fe900, incoming=nullptr) at service_pool.cc:269:19
    frame #18: 0x0000aaaad8cc12c0 yb-master`yb::rpc::InboundCall::InboundCallTask::Run(this=<unavailable>) at inbound_call.cc:303:13
    frame #19: 0x0000aaaad8d81be8 yb-master`yb::rpc::(anonymous namespace)::Worker::Execute(this=<unavailable>) at thread_pool.cc:115:15
    frame #20: 0x0000aaaad9585cb8 yb-master`yb::Thread::SuperviseThread(void*) [inlined] std::__1::__function::__value_func<void ()>::operator()[abi:v170002](this=0x000027953f9576f0) const at function.h:517:16
    frame #21: 0x0000aaaad9585ca4 yb-master`yb::Thread::SuperviseThread(void*) [inlined] std::__1::function<void ()>::operator()(this=0x000027953f9576f0) const at function.h:1168:12
    frame #22: 0x0000aaaad9585ca4 yb-master`yb::Thread::SuperviseThread(arg=0x000027953f957690) at thread.cc:866:3
    frame #23: 0x0000ffff8a8178b8 libpthread.so.0`start_thread + 392
    frame #24: 0x0000ffff8a873afc libc.so.6`thread_start + 12
  thread #2, stop reason = signal 0
    frame #0: 0x0000ffff8a9241e8 libc.so.6`epoll_pwait + 136
    frame #1: 0x0000aaaad7e7a238 yb-master`epoll_poll + 88
    frame #2: 0x0000aaaad7e7c9bc yb-master`ev_run + 1596
    frame #3: 0x0000aaaad8ca6bc8 yb-master`yb::rpc::Acceptor::RunThread() [inlined] ev::loop_ref::run(this=<unavailable>, flags=0) at ev++.h:211:7
    frame #4: 0x0000aaaad8ca6bc0 yb-master`yb::rpc::Acceptor::RunThread(this=<unavailable>) at acceptor.cc:221:9
    frame #5: 0x0000aaaad9585cb8 yb-master`yb::Thread::SuperviseThread(void*) [inlined] std::__1::__function::__value_func<void ()>::operator()[abi:v170002](this=0x000027953f9569e0) const at function.h:517:16
    frame #6: 0x0000aaaad9585ca4 yb-master`yb::Thread::SuperviseThread(void*) [inlined] std::__1::function<void ()>::operator()(this=0x000027953f9569e0) const at function.h:1168:12
    frame #7: 0x0000aaaad9585ca4 yb-master`yb::Thread::SuperviseThread(arg=0x000027953f956980) at thread.cc:866:3
    frame #8: 0x0000ffff8a8178b8 libpthread.so.0`start_thread + 392
    frame #9: 0x0000ffff8a873afc libc.so.6`thread_start + 12
  thread #3, stop reason = signal 0
  .

```

G-flags:
```
master_gflags={
                    "ysql_yb_ddl_rollback_enabled": True,
                    "allowed_preview_flags_csv": "ysql_yb_ddl_rollback_enabled",
                    "ysql_transaction_bg_task_wait_ms": ysql_transaction_bg_task_wait_ms,
                    **disable_guradrail_gflags,
                    **config.additional_master_gflags
                },
tserver_gflags={
                    "ysql_yb_ddl_rollback_enabled": True,
                    "allowed_preview_flags_csv": "ysql_yb_ddl_rollback_enabled",
                    # let CM bg task poll for txn status
                    "report_ysql_ddl_txn_status_to_master": report_ysql_ddl_txn_status_to_master,
                    "ysql_catalog_preload_additional_tables": True,
                    "ysql_enable_read_request_caching": True,
                    **disable_guradrail_gflags,
                    **config.additional_tserver_gflags
                }
```

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-10807]: https://yugabyte.atlassian.net/browse/DB-10807?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by rthallamko3** [comment]
@shishir2001-yb , Can you confirm if this issue is being hit after we addressed #21877 ? If not, lets just close this.

**Comment by shishir2001-yb** [comment]
No longer seeing this issue after the above fix.
